images/win/scripts/SoftwareReport/SoftwareReport.Generator.ps1

```diff
@@ -15,279 +17,226 @@ Import-Module (Join-Path $PSScriptRoot "SoftwareReport.Java.psm1") -DisableNameC
-$markdown += New-MDHeader "Rust Tools" -Level 3
-$markdown += New-MDList -Style Unordered -Lines (@(
-    "Rust $(Get-RustVersion)",
-    "Rustup $(Get-RustupVersion)",
-    "Cargo $(Get-RustCargoVersion)",
-    "Rustdoc $(Get-RustdocVersion)"
-    ) | Sort-Object
-)
-
-$markdown += New-MDHeader "Packages" -Level 4
-$markdown += New-MDList -Style Unordered -Lines (@(
-    (Get-BindgenVersion),
-    (Get-CargoAuditVersion),
-    (Get-CargoOutdatedVersion),
-    (Get-CbindgenVersion),
-    "Rustfmt $(Get-RustfmtVersion)",
-    "Clippy $(Get-RustClippyVersion)"
-    ) | Sort-Object
-)
-
-$markdown += New-MDHeader "Browsers and Drivers" -Level 3
-$markdown += New-MDList -Style Unordered -Lines @(
-    (Get-BrowserVersion -Browser "chrome"),
-    (Get-SeleniumWebDriverVersion -Driver "chrome"),
-    (Get-BrowserVersion -Browser "edge"),
-    (Get-SeleniumWebDriverVersion -Driver "edge"),
-    (Get-BrowserVersion -Browser "firefox"),
-    (Get-SeleniumWebDriverVersion -Driver "firefox"),
-    (Get-SeleniumWebDriverVersion -Driver "iexplorer"),
-    (Get-SeleniumVersion)
-)
-
-$markdown += New-MDHeader "Environment variables" -Level 4
-$markdown += Build-BrowserWebdriversEnvironmentTable | New-MDTable
-$markdown += New-MDNewLine
-
-$markdown += New-MDHeader "Java" -Level 3
-$markdown += Get-JavaVersions | New-MDTable
-$markdown += New-MDNewLine
-
-$markdown += New-MDHeader "Shells" -Level 3
-$markdown += Get-ShellTarget
-$markdown += New-MDNewLine
-
-$markdown += New-MDHeader "MSYS2" -Level 3
-$markdown += "$(Get-PacmanVersion)" | New-MDList -Style Unordered
-$markdown += New-MDHeader "Notes:" -Level 5
-$reportMsys64 = @'
-```
+$cliTools.AddToolVersion("GitHub CLI", $(Get-GHVersion))
+$cliTools.AddToolVersion("Hub CLI", $(Get-HubVersion))
+
+# Rust Tools
+$rustTools = $installedSoftware.AddHeader("Rust Tools")
+$rustTools.AddToolVersion("Cargo", $(Get-RustCargoVersion))
+$rustTools.AddToolVersion("Rust", $(Get-RustVersion))
+$rustTools.AddToolVersion("Rustdoc", $(Get-RustdocVersion))
+$rustTools.AddToolVersion("Rustup", $(Get-RustupVersion))
+
+$rustToolsPackages = $rustTools.AddHeader("Packages")
+$rustToolsPackages.AddToolVersion("bindgen", $(Get-BindgenVersion))
+$rustToolsPackages.AddToolVersion("cargo-audit", $(Get-CargoAuditVersion))
+$rustToolsPackages.AddToolVersion("cargo-outdated", $(Get-CargoOutdatedVersion))
+$rustToolsPackages.AddToolVersion("cbindgen", $(Get-CbindgenVersion))
+$rustToolsPackages.AddToolVersion("Clippy", $(Get-RustClippyVersion))
+$rustToolsPackages.AddToolVersion("Rustfmt", $(Get-RustfmtVersion))
+
+# Browsers and webdrivers
+$browsersAndWebdrivers = $installedSoftware.AddHeader("Browsers and webdrivers")
+$browsersAndWebdrivers.AddNodes($(Build-BrowserSection))
+$browsersAndWebdrivers.AddNode($(Build-BrowserWebdriversEnvironmentTable))
+
+# Java
+$java = $installedSoftware.AddHeader("Java")
+$java.AddTable($(Get-JavaVersions))
+
+# Shells
+$java = $installedSoftware.AddHeader("Shells")
+$java.AddTable($(Get-ShellTarget))
```

Comment:
I think `java` is incorrect variable name here. Rename or get rid of variable at all.

Reply:
got rid of the variable :)